client/core/trade.go

```diff
@@ -1224,6 +1256,83 @@ func (t *trackedTrade) finalizeRedeemAction(match *matchTracker, coinID []byte)
 	return errs.ifAny()
 }
 
+func (t *trackedTrade) beginFindRedemption(match *matchTracker) {
+	swapCoinID := dex.Bytes(match.MetaData.Proof.TakerSwap)
+	fromAsset := t.wallets.fromAsset.ID
+	resultChan, err := t.wallets.fromWallet.FindRedemption(swapCoinID)
+	if err != nil {
+		log.Errorf("failed to start redemption search for match %v, %s contract %s: %v",
+			match.id, unbip(fromAsset), coinIDString(fromAsset, swapCoinID), err)
+		return
+	}
+	// Redemption search started, start goroutine to monitor result channel.
+	t.awaitingRedemption[swapCoinID.String()] = match.id
+	go t.waitForRedemption(resultChan)
+}
+
+// waitForRedemption waits on the passed `redemptionChan` for FindRedemption
+// results. Should be run as a goroutine.
+func (t *trackedTrade) waitForRedemption(redemptionChan chan *asset.FindRedemptionResult) {
+	frr := <-redemptionChan
+	if frr == nil {
+		return // channel closed without data
+	}
```

Comment:
I see now why the channel is closed in `(asset).FindRedeption`, but it seems odd that we should make it here if the `t.awaitingRedemption` is ensuring multiple consumers are not waiting. Could you log a Debug here, so we know if it happens?

Also, should there be some timeout functioning somewhere?

Reply:
This happens when dexc is shutdown before the contract's redemption is found. The asset's shutdown method closes all open redemption channels which causes a nil value to be received here. Not sure what to log here. `Redemption not found for contract on shutdown`?

Reply:
Ah, on shutdown, that makes sense. I thought some other time during operation.